org.code.toboggan.filesystem/src/org/code/toboggan/filesystem/extensions/file/FSFileMove.java

```diff
@@ -0,0 +1,20 @@
+package org.code.toboggan.filesystem.extensions.file;
+
+import java.nio.file.Path;
+
+import org.code.toboggan.core.api.APIFactory;
+import org.code.toboggan.network.request.extensionpoints.file.IFileMoveResponse;
+
+public class FSFileMove implements IFileMoveResponse {
+
+	@Override
+	public void fileMoved(long fileID, Path newWorkspaceRelativePath) {
+		new Thread(APIFactory.createFilePullDiffSendChanges(fileID)).start();
+	}
+
+	@Override
+	public void fileMoveFailed(long fileID, Path newWorkspaceRelativePath) {
+
```

Comment:
I believe you need to try to undo the file move here. The only time this function will be called is if the user moves a file on the filesystem, the directorywatching detects it, and we send a filemoverequest. Then this response extension will get run when the move fails, so we need to move it back. This might require sending in the old path as the parameter.